### DISCUSSION TITLE
How to create ShortcutsWindow with shortcuts programatically(not ui)?

### DISCUSSION THREAD

**Discussion by Wanpaku** [gtk-rs/gtk4-rs#1724] [Q&A] [answered]
I'm not understand how to add ShortcutsGroup to ShortcutsWindow
and how to add ShortcutsShortcut to ShortcutsGroup programatically.
There isn't any method such as "set_child", "child" or "add_group".

**Comment by bilelmoussaoui** [comment]
There is https://gtk-rs.org/gtk4-rs/stable/latest/docs/gtk4/struct.ShortcutsWindow.html#method.add_section

**Reply by Wanpaku** [reply]
It's only for 4.14 and higher version of the crate.
I found that I can add ShortcutsGroup to ShortcutsWindow with "child" method using "builder()".
But I still don't understand how to add ShortcutsShorcut to ShortcutsGroup.

**Reply by bilelmoussaoui** [reply, marked as answer]
Yes, for a long time the only way to create those was using XML. So you need 4.14 unfortunately
